modules/aca/office_booking.rb

```diff
@@ -149,230 +73,78 @@ def on_update
-        end
-
-        # Do we want to look up the users email address?
-        if CAN_LDAP
-            @ldap_creds = setting(:ldap_creds)
-            if @ldap_creds
-                encrypt = @ldap_creds[:encryption]
-                encrypt[:method] = encrypt[:method].to_sym if encrypt && encrypt[:method]
-                @tree_base = setting(:tree_base)
-                @ldap_user = @ldap_creds.delete :auth
-            end
-        else
-            logger.warn "net/ldap gem not available" if setting(:ldap_creds)
-        end
-
-        # Do we want to use exchange web services to manage bookings
-        if CAN_OFFICE
-            logger.debug "Setting OFFICE"
-            @office_organiser_location = setting(:office_organiser_location)
-            @office_client_id = setting(:office_client_id)
-            @office_secret = setting(:office_secret)
-            @office_scope = setting(:office_scope)
-            @office_site = setting(:office_site)
-            @office_token_url = setting(:office_token_url)
-            @office_options = setting(:office_options)
-            @office_user_email = setting(:office_user_email)
-            @office_user_password = setting(:office_user_password)
-            @office_delegated = setting(:office_delegated)
-            @office_room = (setting(:office_room) || system.email)
-            # supports: SMTP, PSMTP, SID, UPN (user principle name)
-            # NOTE:: Using UPN we might be able to remove the LDAP requirement
-            @office_connect_type = (setting(:office_connect_type) || :SMTP).to_sym
-            @timezone = setting(:room_timezone)
-
-            @client = ::Microsoft::Office.new({
-                client_id: @office_client_id || ENV['OFFICE_CLIENT_ID'],
-                client_secret: @office_secret || ENV["OFFICE_CLIENT_SECRET"],
-                app_site: @office_site || ENV["OFFICE_SITE"] || "https://login.microsoftonline.com",
-                app_token_url: @office_token_url || ENV["OFFICE_TOKEN_URL"],
-                app_scope: @office_scope || ENV['OFFICE_SCOPE'] || "https://graph.microsoft.com/.default",
-                graph_domain: ENV['GRAPH_DOMAIN'] || "https://graph.microsoft.com",
-                service_account_email: @office_user_password || ENV['OFFICE_ACCOUNT_EMAIL'],
-                service_account_password: @office_user_password || ENV['OFFICE_ACCOUNT_PASSWORD'],
-                internet_proxy: @internet_proxy || ENV['INTERNET_PROXY']
-            })
-        else
-            logger.warn "oauth2 gem not available" if setting(:office_creds)
-        end
-
-        # Load the last known values (persisted to the DB)
-        self[:waiter_status] = (setting(:waiter_status) || :idle).to_sym
-        self[:waiter_call] = self[:waiter_status] != :idle
-
-        self[:catering_status] = setting(:last_catering_status) || {}
-        self[:order_status] = :idle
+        logger.debug "Setting OFFICE"
+        @office_client_id = setting(:office_client_id)
```

Comment:
This and other credentials / access details are not used anywhere outside of the client creation below. These can probably be switched to local vars rather than client variables so that cleartext credentials are not sitting in memory for any longer than needed.

Reply:
agreed, resolved.